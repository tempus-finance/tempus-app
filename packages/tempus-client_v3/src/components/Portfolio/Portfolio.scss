.tc__app__portfolio {
  flex-grow: 1;
}

.tc__app__portfolio-content {
  padding: 40px 80px 80px 80px;
}

.tc__app__portfolio-box {
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  border: 1px solid #bfc0c0;
  border-radius: 16px;
  box-sizing: border-box;
}

.tc__app__portfolio-info-container {
  display: flex;
  margin-bottom: 20px;

  .tc__app__portfolio-info-box {
    padding: 24px;
    flex-grow: 1;

    &:not(:last-child) {
      margin-right: 12px;
    }

    .tc__app__portfolio-info-box-value {
      margin-top: 20px;
      display: flex;
    }

    .tc__app__portfolio-info-box-currency {
      margin-left: 12px;
    }
  }
}

.tc__app__portfolio-main-content {
  padding: 28px 24px 24px 24px;

  .tc__app__portfolio-header {
    margin-bottom: 28px;
    display: flex;
    justify-content: space-between;

    .tc__tabs__tab {
      width: 104px;
    }
  }
}
