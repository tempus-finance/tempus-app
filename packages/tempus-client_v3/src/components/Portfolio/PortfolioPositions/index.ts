import PortfolioPositions from './PortfolioPositions';

export default PortfolioPositions;
