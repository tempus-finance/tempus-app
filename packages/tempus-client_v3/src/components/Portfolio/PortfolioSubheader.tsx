import { FC, useCallback, useState } from 'react';
import { useTranslation } from 'react-i18next';
import { NavSubheader, NavSubheaderGroup, Tab, Tabs } from '../shared';

export type PortfolioView = 'overview' | 'positions';

export interface PortfolioSubheaderProps {
  onViewChange?: (view: PortfolioView) => void;
}

const PortfolioSubheader: FC<PortfolioSubheaderProps> = props => {
  const { onViewChange } = props;
  const [selectedView, setSelectedView] = useState<PortfolioView>('overview');
  const { t } = useTranslation();

  const handleViewChange = useCallback(
    (view: PortfolioView) => {
      setSelectedView(view);
      onViewChange?.(view);
    },
    [onViewChange],
  );

  return (
    <NavSubheader>
      <NavSubheaderGroup>
        <Tabs size="small" value={selectedView} onTabSelected={handleViewChange}>
          <Tab label={t('PortfolioSubheader.tabOverview')} value="overview" />
          <Tab label={t('PortfolioSubheader.tabPositions')} value="positions" />
        </Tabs>
      </NavSubheaderGroup>
    </NavSubheader>
  );
};

export default PortfolioSubheader;
