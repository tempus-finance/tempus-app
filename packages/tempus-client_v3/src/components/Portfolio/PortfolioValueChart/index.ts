import PortfolioValueChart from './PortfolioValueChart';

export default PortfolioValueChart;
