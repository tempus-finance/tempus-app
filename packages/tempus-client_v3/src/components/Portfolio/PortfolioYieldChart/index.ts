import PortfolioYieldChart from './PortfolioYieldChart';

export default PortfolioYieldChart;
