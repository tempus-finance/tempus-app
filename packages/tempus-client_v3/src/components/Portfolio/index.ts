import Portfolio from './Portfolio';
import PortfolioOverview from './PortfolioOverview';
import PortfolioPositions from './PortfolioPositions';

export { PortfolioPositions, PortfolioOverview };
export default Portfolio;
