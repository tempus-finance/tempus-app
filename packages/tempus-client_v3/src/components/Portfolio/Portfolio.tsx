import { FC, useState } from 'react';
import PortfolioOverview from './PortfolioOverview';
import PortfolioSubheader, { PortfolioView } from './PortfolioSubheader';
import './Portfolio.scss';

const Portfolio: FC = () => {
  const [view, setView] = useState<PortfolioView>('overview');

  return (
    <div className="tc__app__portfolio">
      <PortfolioSubheader onViewChange={setView} />
      <div className="tc__app__portfolio-content">{view === 'overview' && <PortfolioOverview />}</div>
    </div>
  );
};

export default Portfolio;
