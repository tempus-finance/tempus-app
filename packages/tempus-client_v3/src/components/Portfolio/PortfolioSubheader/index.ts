import PortfolioSubheader from './PortfolioSubheader';

export default PortfolioSubheader;
export * from './PortfolioSubheader';
