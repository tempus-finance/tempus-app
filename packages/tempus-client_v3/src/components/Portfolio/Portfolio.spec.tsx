import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import Portfolio from './Portfolio';

const subject = () =>
  render(
    <BrowserRouter>
      <Portfolio />
    </BrowserRouter>,
  );

describe('Portfolio', () => {
  it('renders portfolio page', () => {
    const { container } = subject();

    expect(container).not.toBeNull();
    expect(container).toMatchSnapshot();
  });
});
