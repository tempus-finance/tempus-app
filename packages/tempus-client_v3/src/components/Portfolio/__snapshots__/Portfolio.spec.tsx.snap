// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Portfolio renders portfolio page 1`] = `
<div>
  <div
    class="tc__app__portfolio"
  >
    <div
      class="tc__nav-subheader"
    >
      <div
        class="tc__nav-subheader__background"
      />
      <div
        class="tc__nav-subheader__wrapper"
      >
        <div
          class="tc__nav-subheader__group"
        >
          <div
            class="tc__tabs tc__tabs__tabs-small"
          >
            <button
              class="tc__btn tc__tabs__tab"
              data-selected="true"
              type="button"
            >
              <div
                style="font-style: normal; font-size: 12px; line-height: 16px; color: rgb(255, 255, 255); font-weight: 700; font-family: 'DM Sans', sans-serif; opacity: 1;"
              >
                Overview
              </div>
            </button>
            <button
              class="tc__btn tc__tabs__tab"
              data-selected="false"
              type="button"
            >
              <div
                style="font-style: normal; font-size: 12px; line-height: 16px; color: rgb(34, 34, 34); font-weight: 500; font-family: 'DM Sans', sans-serif; opacity: 1;"
              >
                Positions
              </div>
            </button>
            <div
              class="tc__tabs__indicator-container"
            >
              <div
                class="tc__tabs__indicator"
                style="width: 50%; left: 0%;"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="tc__app__portfolio-content"
    >
      <div>
        <div
          class="tc__app__portfolio-info-container"
        >
          <div
            class="tc__app__portfolio-box tc__app__portfolio-info-box"
          >
            <div
              style="font-style: normal; font-size: 24px; line-height: 32px; color: rgb(34, 34, 34); font-weight: 700; font-family: 'DM Sans', sans-serif; opacity: 1;"
            >
              Current Value
            </div>
            <div
              style="font-style: normal; font-size: 20px; line-height: 28px; color: rgb(34, 34, 34); font-weight: 400; font-family: 'DM Sans', sans-serif; opacity: 1;"
            >
              of portfolio
            </div>
            <div
              class="tc__app__portfolio-info-box-value"
            >
              <div
                style="font-style: normal; font-size: 32px; line-height: 40px; color: rgb(34, 34, 34); font-weight: 500; font-family: 'Azeret Mono', monospace; opacity: 1;"
              >
                123.45
              </div>
              <div
                class="tc__app__portfolio-info-box-currency"
                style="font-style: normal; font-size: 32px; line-height: 40px; color: rgb(71, 67, 80); font-weight: 500; font-family: 'DM Sans', sans-serif; opacity: 1;"
              >
                USD
              </div>
            </div>
          </div>
          <div
            class="tc__app__portfolio-box tc__app__portfolio-info-box"
          >
            <div
              style="font-style: normal; font-size: 24px; line-height: 32px; color: rgb(34, 34, 34); font-weight: 700; font-family: 'DM Sans', sans-serif; opacity: 1;"
            >
              Yield Earned
            </div>
            <div
              style="font-style: normal; font-size: 20px; line-height: 28px; color: rgb(34, 34, 34); font-weight: 400; font-family: 'DM Sans', sans-serif; opacity: 1;"
            >
              of portfolio
            </div>
            <div
              class="tc__app__portfolio-info-box-value"
            >
              <div
                style="font-style: normal; font-size: 32px; line-height: 40px; color: rgb(34, 34, 34); font-weight: 500; font-family: 'Azeret Mono', monospace; opacity: 1;"
              >
                30.24
              </div>
              <div
                class="tc__app__portfolio-info-box-currency"
                style="font-style: normal; font-size: 32px; line-height: 40px; color: rgb(71, 67, 80); font-weight: 500; font-family: 'DM Sans', sans-serif; opacity: 1;"
              >
                USD
              </div>
            </div>
          </div>
          <div
            class="tc__app__portfolio-box tc__app__portfolio-info-box"
          >
            <div
              style="font-style: normal; font-size: 24px; line-height: 32px; color: rgb(34, 34, 34); font-weight: 700; font-family: 'DM Sans', sans-serif; opacity: 1;"
            >
              Projected Yield
            </div>
            <div
              style="font-style: normal; font-size: 20px; line-height: 28px; color: rgb(34, 34, 34); font-weight: 400; font-family: 'DM Sans', sans-serif; opacity: 1;"
            >
              of portfolio
            </div>
            <div
              class="tc__app__portfolio-info-box-value"
            >
              <div
                style="font-style: normal; font-size: 32px; line-height: 40px; color: rgb(34, 34, 34); font-weight: 500; font-family: 'Azeret Mono', monospace; opacity: 1;"
              >
                42.78
              </div>
              <div
                class="tc__app__portfolio-info-box-currency"
                style="font-style: normal; font-size: 32px; line-height: 40px; color: rgb(71, 67, 80); font-weight: 500; font-family: 'DM Sans', sans-serif; opacity: 1;"
              >
                USD
              </div>
            </div>
          </div>
        </div>
        <div
          class="tc__app__portfolio-box tc__app__portfolio-main-content"
        >
          <div
            class="tc__app__portfolio-header"
          >
            <div
              style="font-style: normal; font-size: 24px; line-height: 32px; color: rgb(34, 34, 34); font-weight: 700; font-family: 'DM Sans', sans-serif; opacity: 1;"
            >
              Total Yield Earned
            </div>
            <div
              class="tc__tabs tc__tabs__tabs-small"
            >
              <button
                class="tc__btn tc__tabs__tab"
                data-selected="true"
                type="button"
              >
                <div
                  style="font-style: normal; font-size: 12px; line-height: 16px; color: rgb(255, 255, 255); font-weight: 700; font-family: 'DM Sans', sans-serif; opacity: 1;"
                >
                  Total Yield
                </div>
              </button>
              <button
                class="tc__btn tc__tabs__tab"
                data-selected="false"
                type="button"
              >
                <div
                  style="font-style: normal; font-size: 12px; line-height: 16px; color: rgb(34, 34, 34); font-weight: 500; font-family: 'DM Sans', sans-serif; opacity: 1;"
                >
                  Total Value
                </div>
              </button>
              <div
                class="tc__tabs__indicator-container"
              >
                <div
                  class="tc__tabs__indicator"
                  style="width: 50%; left: 0%;"
                />
              </div>
            </div>
          </div>
          <div
            class="recharts-responsive-container"
            style="width: 100%; height: 512px;"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
